File: src/programs/conditionalsAndLoops/eight.js


import '../../styles/programs.css'
import '../../styles/topics.css'
import ConditionalsAndLoops from '../../topics/ConditionalsAndLoops'
import { useState,useEffect } from 'react'
import ConditionalsAndLoopsNine from './nine'
const ConditionalsAndLoopsEight = ({qn,backPage}) =>{
    const q9 = "Calculate Discount Of Product"
    const back = (page) =>{
        backPage(page)
    }
        const url = window.location.href

        const [file,setFile] = useState("")
    
        useEffect(()=>{
        async function fetchFile() {
            try{
        const res = await fetch(`${url}javaPrograms/conditionalsAndLoops/Eight.java`)
   
        const cont = await res.text();
        
        setFile(cont)
        }
        catch(err){
        console.log(err);
        }

    }
     
    fetchFile() 
  },[])
    return(<div className="program-body">
        <div className='topics-logo'>{qn}</div>
        <div className='program-content'>

          
            <div className='program-hint program'>
                <pre>  {file}</pre>
         

            </div>
            <div className='program-btns'>
                
                </div>
            <div onClick={() => back(<ConditionalsAndLoops/>)} className='program-btn'>Go To Map</div>
            <div onClick={() => back(<ConditionalsAndLoopsNine qn ={q9} backPage={backPage}/>)} className='program-btn'>Next</div>
        </div>
    
    </div>)
}
export default ConditionalsAndLoopsEight

File: public/javaPrograms/conditionalsAndLoops/Eight.java
//Calculate Average Of N Numbers
package javaPrograms.conditionalsAndLoops;

import java.util.Scanner;

public class Eight {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
     
     
    int add = 0,i;
    System.out.print("enter limit:");
    int n =s.nextInt();
    for(i = 1;i<=n;i++){
        System.out.print("Enter "+i+" no:"); 
        int no = s.nextInt();
        add = add+no;
    }
    System.out.println("average :"+(add/n));
    
   
            
      
    }
   

}
